File: Data Structures & Algorithms/string-encode-and-decode/submission-2.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        encoding_character = '#'
        encoded_str = ''
        for word in strs:
            encoded_str+=str(len(word)) + encoding_character + word
        return encoded_str


    def decode(self, s: str) -> List[str]:
        result = []
        print(s)
        i = 0
        temp = ''
        while i<len(s):
            if s[i] == '#':
                length = int(temp)
                result.append(s[i+1:i+length+1])
                temp=''
                i+=length+1
            else:
                temp+=s[i]
                i+=1
        return result

File: Data Structures & Algorithms/anagram-groups/submission-2.cpp
class Solution {
   public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        map<vector<int>, vector<string>> anagrams;

        for (string str : strs) {
            vector<int> vect(26,0);

            for (char x : str) {
                vect[int(x) - int('a')] += 1;
            }

            anagrams[vect].push_back(str);
        }

        vector<vector<string>> result;

        for (const auto& [key, value] : anagrams) {
            result.push_back(value);
        }

        return result;
    }
};
